patches/configs/310.config

```diff
@@ -1,8 +1,7 @@
 [build_id]
-main=034C8FA7A63B7A87F96F408B2AEFFF6C
-sdk=EAD5AAD43C3EFA01682460ECE70A8189F84D2AFF
+main=3CA12DFAAF9C82DA
 
 [nso_load_addr]
 main=0
-subsdk0=0x48DF000
-;sdk=subsdk0+1 TODO: need load address formula to calculate
+subsdk0=0x25A8000
+subsdk1=0x2C63000
```

Comment:
These offsets represent where each module will be loaded, relative to `main`.

Vanilla SMO will load four modules, in this order in memory:

1. `rtld`
2. `main`
3. `subsdk0`
4. `sdk`

Our goal is to add a fifth: `subsdk1`. This will be loaded in memory after `subsdk0` and before `sdk`.

The easiest way I've found to obtain these offsets is to add any `subsdk1` module to the game (I used the output from the original, Splatoon 2-based Starlight), and then to cause a crash (I used a patch file than contained the instructions `MOV X0, #0; STR X0, [X0]`). Atmosphère's crash report will include information about which modules are loaded, including their load addresses; you can then subtract `main`'s load address from each to get each module's relative load address. There is likely a way to get these offsets statically — I just do not know it.

The offsets are necessary in order to know where to branch through within our patches; for example, if we wanted to branch from relative offset `0x1000` in main, to relative offset `0x2000` in subsdk1 (or new module), then we could compute the distance between them as `0x2C64000`, and use `BL 0x2C64000` accordingly:

```
(subsdk1_load_addr + subsdk1_relative_offset) - (main_load_addr + main_relative_offset)`
= (0x2C63000 + 0x2000) - (0x0 + 0x1000)`
= 0x2C64000
```

Note that we do not need `rtld` or `sdk`'s offsets here; we only need offsets of modules we plan to branch to.
If we do need to branch to `sdk` in the future, the address will likely need to be computed separately (since it will be dependent on the size of `subsdk1`).